nx/include/switch/services/ins.h

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file ins.h
+ * @brief INS services IPC wrapper.
+ * @author averne
+ * @copyright libnx Authors
+ */
+#pragma once
+#include "../types.h"
+#include "../kernel/event.h"
+#include "../sf/service.h"
+
+/// Initialize ins:r.
+Result insrInitialize(void);
+
+/// Exit ins:r.
+void insrExit(void);
+
+/// Gets the Service object for the actual ins:r service session.
+Service* insrGetServiceSession(void);
+
+/**
+ * @brief Retrieves the last system tick the event corresponding to the ID was signaled at.
+ * @param[in] id Ins request ID (should be 0..4).
+ * @param[out] tick.
+ * @return Result code.
+ * @note The tick is only updated once per second at minimum.
+ */
+Result insrGetLastTick(u32 id, u64 *tick);
```

Comment:
Is NN known to use an enum for this ID, or is it really just a plain integer?

Reply:
Presumably, yes.
I looked a bit more into this, IDs are mapped as follows:
- InsRequestId 0 -> InsSendId 0-11
- InsRequestId 1 -> InsSendId 0, 2-9
- InsRequestId 2 -> InsSendId 0, 2, 4-9
- InsRequestId 3 -> InsSendId 0-9, 11
- InsRequestId 4 -> InsSendId 1-9, 11

4 services retrieve ins:s write handles:
- USB: InsSendId 6
- HID: InsSendId 0-5, 9-11
- Audio: InsSendId 0
- NS: InsSendId 3, 7

Only AM subscribes to ins:r read handles, with IDs 0, 2 & 3.

From this it's not really clear to me what they are supposed to represent. I'd say low InsSendIds are used to signal activity of critical software components, while high IDs represent specific hardware (eg BT/NFC in the case of hid)?